src/inference/dev_api/openvino/runtime/threading/istreams_executor.hpp

```diff
@@ -173,32 +115,116 @@ class OPENVINO_RUNTIME_API IStreamsExecutor : virtual public ITaskExecutor {
+            return _name;
+        }
+        int get_streams() {
+            return _streams;
+        }
+        int get_streams() const {
+            return _streams;
+        }
+        int get_threads() {
+            return _threads;
+        }
+        int get_threads() const {
+            return _threads;
+        }
+        int get_threads_per_stream() {
+            return _threads_per_stream;
+        }
+        bool get_cpu_reservation() {
+            return _cpu_reservation;
+        }
+        std::vector<std::vector<int>> get_streams_info_table() {
+            return _streams_info_table;
+        }
+        std::vector<std::vector<int>> get_stream_processor_ids() {
+            return _stream_processor_ids;
+        }
+        ThreadBindingType get_thread_binding_type() {
+            return _threadBindingType;
+        }
+        int get_thread_binding_step() {
+            return _threadBindingStep;
+        }
+        int get_thread_binding_offset() {
+            return _threadBindingOffset;
+        }
+        bool compare(Config config) {
+            if (_name == config._name && _streams == config._streams &&
+                _threads_per_stream == config._threads_per_stream && _threadBindingType == config._threadBindingType &&
+                _thread_preferred_core_type == config._thread_preferred_core_type) {
+                return true;
+            } else {
+                return false;
+            }
+        }
+
+        /**
+         * @brief Create appropriate multithreaded configuration
+         *        filing unconfigured values from initial configuration using hardware properties
+         * @param initial Inital configuration
+         * @return configured values
+         */
+        static Config make_default_multi_threaded(const Config& initial);
+
+        static int get_default_num_streams();  // no network specifics considered (only CPU's caps);
+
+        /**
+         * @brief Get and reserve cpu ids based on configuration and hardware information
+         *        streams_info_table must be present in the configuration
+         * @param initial Inital configuration
+         * @return configured values
+         */
+        // It will be removed when other plugins will no longer call it.
+        static Config reserve_cpu_threads(const Config& initial);
+
+        /**
+         * @brief Modify _streams_info_table and related configuration according to configuration
+         */
+        void update_executor_config();
+
         /**
          * @brief Set _streams_info_table and _cpu_reservation in cpu streams executor config when nstreams = 0,
          *        that is, only create one thread with TBB
+         * @param streamsInfoTable streams info table
+         * @param cpu_reservation cpu reservation
          */
-        void set_config_zero_stream();
+        static void set_config_zero_stream(std::vector<std::vector<int>>& streamsInfoTable, bool& cpu_reservation);
```

Comment:
Why it became static method with some params? It should just change the state of the config like before.

Reply:
Because it was called in cpu plugin before. Now it is moved into `update_executor_config`, so changed it to private.